src/js/stores/PluginStore.js

```diff
@@ -16,7 +16,9 @@ var storeData = {
 };
 
 function addPlugin(data) {
-  if (data.id == null && !Util.isObject(data.info) &&
+  if (data.id == null || !Util.isObject(data.info) ||
+      !Util.isArray(data.info.modules) ||
+      !data.info.modules.includes(PluginModules.UI) ||
```

Comment:
Your call – but I do see some benefit in that we could unit test that particular function.